datafusion/core/tests/physical_optimizer/enforce_sorting.rs

```diff
@@ -1943,6 +1943,30 @@ async fn test_remove_unnecessary_spm1() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn test_remove_unnecessary_spm2() -> Result<()> {
+    let schema = create_test_schema()?;
+    let source = memory_exec(&schema);
+    let input = sort_preserving_merge_exec_with_fetch(
+        vec![sort_expr("non_nullable_col", &schema)],
+        source,
+        100,
+    );
+
+    let expected_input = [
+        "SortPreservingMergeExec: [non_nullable_col@1 ASC], fetch=100",
+        "  DataSourceExec: partitions=1, partition_sizes=[0]",
+    ];
+    let expected_optimized = [
+        "SortPreservingMergeExec: [non_nullable_col@1 ASC], fetch=100",
+        "  SortExec: expr=[non_nullable_col@1 ASC], preserve_partitioning=[false]",
```

Comment:
I am wandering do we expect to optimize only sortExec with Topk(fetch) for the only one partition case?